src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/ComponentDatabase.java

```diff
@@ -9,8 +10,25 @@
     return component == null ? null : component.getParent();
   }
 
-  public static @Nullable Component getEditorWindow() {
-    return ComponentLocator.getComponentByClass("EditorWindow");
+  /**
+   * Returns any of the open editors - it is not deterministic which one will be returned.
+   */
+  public static @Nullable EditorComponentImpl getEditorWindow() {
+    return (EditorComponentImpl) ComponentLocator.getComponentByClass("EditorComponentImpl");
+  }
+
+  /**
+   * Out of all open editors, retrieves the one with a specific file open.
+   */
+  public static @Nullable EditorComponentImpl getEditorWindow(String fileNameSubstring) {
```

Comment:
Why file name substring? Sounds like a source of (user) errors?

Reply:
@Taikelenn